### ISSUE TITLE
Support PROXY protocol for HTTP health checks

### ISSUE THREAD

**Issue by joshuaspence** [hashicorp/consul#3926]
We are using Consul to expose a HTTP service that is backed by nginx. Nginx is listening on ports 80 and 443 using [`PROXY` protocol](https://docs.nginx.com/nginx/admin-guide/load-balancer/using-proxy-protocol/). Ideally we would configure a `http` health check for this service, but we can't do so because the `http` health check doesn't support `PROXY` protocol. Without `PROXY` protocol, the HTTP check fails:

```
Mar 01 21:53:21 ip-10-7-28-60 consul[6248]:     2018/03/01 21:53:21 [WARN] agent: http request failed 'http://127.0.0.1:80': Get http://127.0.0.1:80: EOF
```

**Comment by pearkes** [comment]
I believe if you set `HTTP_PROXY` where Consul is running you should see the behavior you want here. Can you try that?

**Comment by joshuaspence** [comment]
`HTTP_PROXY` has nothing to do with `PROXY` protocol.

**Comment by pearkes** [comment]
Ah I misunderstood and thought you wanted to configure a proxy that would be utilized by Consul while making the request to the HTTP endpoint it is configured to check.

I’m still not totally sure I understand the situation but would you be able to accomplish this with a script check?

**Comment by joshuaspence** [comment]
> I’m still not totally sure I understand the situation but would you be able to accomplish this with a script check?

Of course, but that would involve installing something onto the host that can generate the `PROXY` protocol headers. It would obviously be ideal if Consul could handle this using a library such as https://github.com/armon/go-proxyproto.
